Assets/Scripts/EOSAchievementManager.cs

```diff
@@ -233,19 +151,10 @@ public void Refresh()
         /// The ProductUserId associated with the player for whom the statistics
         /// and achievements are to be updated for.
         /// </param>
-        private void RefreshPlayerStatsAndAchievements(ProductUserId productUserId)
+        private async void RefreshPlayerAchievementsAsync(ProductUserId productUserId)
         {
-            QueryPlayerStats(productUserId, (ref OnQueryStatsCompleteCallbackInfo statsInfo) =>
-            {
-                _playerStats[productUserId] = GetCachedPlayerStats(productUserId);
-            });
-
-            QueryPlayerAchievements(productUserId, (ref OnQueryPlayerAchievementsCompleteCallbackInfo achievementsInfo) =>
-            {
-                _playerAchievements[productUserId] = GetCachedPlayerAchievements(productUserId);
-            });
-
-            NotifyAchievementDataUpdated();
+            _playerAchievements[productUserId] = await QueryPlayerAchievementsAsync(productUserId);
```

Comment:
It _feels_ like we should use something like `AddOrUpdate`, but it's not clear to me if that actually provides significant benefit over what you're doing here.

Reply:
Excellent observation - `AddOrUpdate` is much more appropriate in this context because it is thread-safe, which is the main purpose behind using `ConcurrentDictionary`. I have updated the implementation here, and in the function within `StatsManager` that updates the player stats.